### ISSUE TITLE
Ledger device: Condition of use not satisfied (denied by the user?) (0x6985)

### ISSUE THREAD

**Issue by mudgen** [LedgerHQ/ledgerjs#589]
Hello,

When I try to sign with a Ledger the deployment transaction of the below Ethereum contract I get the following error:

```
Error: 
    at new TransportStatusError (/home/mudgen/projects/aavegotchi/aavegotchi-contracts/node_modules/@ledgerhq/errors/dist/index.cjs.js:289:18)
    at TransportNodeHid.Transport.send (/home/mudgen/projects/aavegotchi/aavegotchi-contracts/node_modules/@ledgerhq/hw-transport/src/Transport.js:212:13)
    at Eth.signTransaction (/home/mudgen/projects/aavegotchi/aavegotchi-contracts/node_modules/@ledgerhq/hw-transport/src/Transport.js:334:16) {
  message: 'Ledger device: Condition of use not satisfied (denied by the user?) (0x6985)',
  statusCode: 27013,
  statusText: 'CONDITIONS_OF_USE_NOT_SATISFIED'
}
```

Here is the source code of the contract:
```
// SPDX-License-Identifier: MIT
pragma solidity 0.8.1;

import {LibDiamond} from "../shared/libraries/LibDiamond.sol";
import {AppStorage} from "./libraries/LibAppStorage.sol";
import {IERC165} from "../shared/interfaces/IERC165.sol";
import {IDiamondCut} from "../shared/interfaces/IDiamondCut.sol";
import {IDiamondLoupe} from "../shared/interfaces/IDiamondLoupe.sol";
import {IERC173} from "../shared/interfaces/IERC173.sol";
import {IERC721} from "../shared/interfaces/IERC721.sol";

contract InitDiamond {
    AppStorage internal s;

    function init(address _rootChainManager) external {
        LibDiamond.DiamondStorage storage ds = LibDiamond.diamondStorage();
        s.itemsBaseUri = "https://aavegotchi.com/metadata/items/";
        s.rootChainManager = _rootChainManager; 

        // adding ERC165 data
        ds.supportedInterfaces[type(IERC165).interfaceId] = true;
        ds.supportedInterfaces[type(IDiamondCut).interfaceId] = true;
        ds.supportedInterfaces[type(IDiamondLoupe).interfaceId] = true;
        ds.supportedInterfaces[type(IERC173).interfaceId] = true;
        ds.supportedInterfaces[type(IERC721).interfaceId] = true;        
    }
}
```

If I modify the source code of the contract and try again the Ledger successfully signs the deployment transaction.  So the problem seems related to the bytecode code generated by this contract or the address generated by this contract.

Maybe the deployment is failing because the contract bytecode is very small for this contract and Ledger doesn't expect that?

Please help.  I don't think Ledger should fail based on the bytecode or address generated from a contract deployment transaction.

**Comment by sterlu** [comment]
We've had numerous reports of users' transactions being rejected again, just like mudgen explained. We saw similar issues last year that seem to have been fixed by https://github.com/LedgerHQ/ledgerjs/pull/514. The behavior has resurfaced for **users that have updated their Ledger firmware to v2**. Pinging @gre (apologies 🙏 ) who seems to be maintaining the app and wrote the PR - seems high priority as people are unable to sing transactions.

**Comment by CrazybutSolid** [comment]
+1.

**Comment by mudgen** [comment]
I have had this problem come up again. Much appreciated if this issue could be addressed.

**Comment by ridiculousgeek** [comment]
+1. Problem still exists on latest Nano X firmware 1.3 / ETH 1.7.7.

**Comment by Jay-Prime** [comment]
Have been seeing this issue with a growing number of users in a number of support channels - transaction never shows up on the Ledger and is reported as instantly denied. All with Ledgers connected directly, though it's not working through MetaMask, either.

This behaviour started with the 2.0.0 update on Nano S and 1.3.0 on Nano X.

Any update would be greatly appreciated.

**Comment by mudgen** [comment]
@gre Sorry to tag you. Is this issue going to be addressed soon?

**Comment by pscott** [comment]
> Have been seeing this issue with a growing number of users in a number of support channels - transaction never shows up on the Ledger and is reported as instantly denied. All with Ledgers connected directly, though it's not working through MetaMask, either.
> 
> This behaviour started with the 2.0.0 update on Nano S and 1.3.0 on Nano X.
> 
> Any update would be greatly appreciated.

Are your users sure they're running the latest versions of the Ethereum app (i.e. 1.7.7 as of today) ?

@mudgen Could you share the bytecode that error'd ? Having some failing transactions would increase the debugging process a lot (currently I can't seem to trigger failing transactions :( )

Thanks!

**Comment by pscott** [comment]
This was a problem in the [app-ethereum](https://github.com/ledgerhq/app-ethereum).

Fixed in [v1.7.8](https://github.com/ledgerhq/app-ethereum/compare/1.7.7...1.7.8) which should be on the live in a couple of hours (deployment...).

Thanks! Closing this one!
